### PR TITLE
CMD accept remote folder explicitly

### PR DESCRIPTION
**Pull request by TheOneRing**
```
Usage: C:\CraftRoot\build\_\30b36579\owncloud-client\out\build\x64-Debug\bin\owncloudcmd.exe [options] source_dir server_url remote_folder
owncloudcmd version 3.1.0.0-git - command line client tool

Options:
  -s, --silten                      Don't be so verbose.
  --httpproxy <http://server:port>  Specify a http proxy to use.
  --trust                           Trust the SSL certification
  --exclude <file>                  Path to an exclude list [file]
  --unsyncedfolders <file>          File containing the list of unsynced remote
                                    folders (selective sync)
  --server <url>                    Use [url] as the location of the server.
                                    OCIS only (server location and spaces url
                                    can differ)
  -u, --user <name>                 Use [name] as the login name
  -p, --password <password>         Use [pass] as password
  -n                                Use netrc (5) for login
  --non-interactive                 Do not block execution with interaction
  --max-sync-retries <n>            Retries maximum n times (default to 3)
  --uplimit <n>                     Limit the upload speed of files to n KB/s
  --downlimit <n>                   Limit the download speed of files to n KB/s
  --logdebug                        More verbose logging
  -?, -h, --help                    Displays help on commandline options.
  --help-all                        Displays help including Qt specific
                                    options.
  -v, --version                     Displays version information.

Arguments:
  source_dir                        The source dir
  server_url                        The url to the server
  remote_folder                     A remote folder
```

# ownCLoud 10
So instead of:
`-u Alice ./tmp https://foo.bar/remote.php/webdav/Alice/Documents`

The user must now provide:
`-u Alice ./tmp https://foo.bar/ Documents`

# ocis
`--server https://ocis.owncloud.test ./tmp https://ocis.owncloud.test/remote.php/dav/spaces/1284d238-aa92-42ce-bdc4-0b0000009157%244c510ada-c86b-4815-8820-42cdf82c3d51 Documents`

### CODE REVIEW

File: changelog/unreleased/10193 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Change: CMD Client, don't guess remote folder
```
Comment (fmoc) [inline changelog/unreleased/10193:1]:
```suggestion
Change: Don't guess remote folder in owncloudcmd
```

File: changelog/unreleased/10193 [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+The commandline client was modified to explicitly accept remote folder, the must no longer be encoded in the server url.
```
Comment (fmoc) [inline changelog/unreleased/10193:3]: I'm not sure what you're trying to say...?

**Review by fmoc — CHANGES_REQUESTED** [review]

File: changelog/unreleased/10193 (hunk 2 above)
Comment (TheOneRing) [reply changelog/unreleased/10193:3]: missing subject

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13604/6/1

:boom: The GUI tests failed.

GUI Logs: https://cache.owncloud.com/public/owncloud/client/13604/guiReportUpload/index.html

<details>

Server Logs: https://cache.owncloud.com/public/owncloud/client/13604/guiReportUpload/serverlog.log

Screenshots:
- [share_file_and_folder_to_a_group.png](https://cache.owncloud.com/public/owncloud/client/13604/guiReportUpload/screenshots/share_file_and_folder_to_a_group.png)

</details>

**Review by fmoc — APPROVED** [review]

**Comment by HanaGemela** [comment]
@TheOneRing provide steps to test

**Comment by TheOneRing** [comment]
> @TheOneRing provide steps to test

https://github.com/owncloud/client/pull/10239#issue-1438150019

The invocation for oc10 and ocis are mentioned.
This change was part of 3.0 and probably tested during change log testing?

**Pull request merged**